Polymer build fails on 2.0 pre2

Polymer build fails to build a project that contains vaadin-grid 2.0 (using latest 0.6 version). 

Problems:

1. Missing HTML imports. Each of the following needs to import polymer or the behavior requested. The UNKNOWN refers to the Edge browser check, that doesn't work.
```
error:   In bower_components/vaadin-grid/vaadin-grid-table-cell.html: [unknown-polymer-behavior] - Unable to resolve behavior `Polymer.Templatizer`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/vaadin-grid-table-cell.html: [unknown-polymer-behavior] - Unable to resolve behavior `Polymer.Templatizer`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/vaadin-grid-table-cell.html: [unknown-polymer-behavior] - Unable to resolve behavior `Polymer.Templatizer`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/vaadin-grid-table-edge-behavior.html: [unknown-polymer-behavior] - Unable to resolve behavior `vaadin.elements.grid.GridTableScrollBehavior`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/iron-list-behavior.html: [unknown-polymer-behavior] - Unable to resolve behavior `Polymer.Templatizer`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/iron-list-behavior.html: [unknown-polymer-behavior] - Unable to resolve behavior `Polymer.IronResizableBehavior`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/iron-list-behavior.html: [unknown-polymer-behavior] - Unable to resolve behavior `Polymer.IronScrollTargetBehavior`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/vaadin-grid-table.html: [unknown-polymer-behavior] - Unable to resolve behavior `UNKNOWN`. Did you import it? Is it annotated with @polymerBehavior?
```

2. iron-list-behavior extends a behavior incorrectly. Should be:
```

 ```/**
 * @polymerBehavior vaadin.elements.grid.IronListBehavior
 */
vaadin.elements.grid.IronListBehaviorImpl = (function() {
  // exactly the same as how you have it, except remove the behaviors property
})();

/**
 * @polymerBehavior vaadin.elements.grid.IronListBehavior
 */
vaadin.elements.grid.IronListBehavior = [
  Polymer.Templatizer,
  Polymer.IronResizableBehavior,
  Polymer.IronScrollTargetBehavior,
  vaadin.elements.grid.IronListBehaviorImpl
];
```
See: https://www.polymer-project.org/1.0/docs/devguide/behaviors#extending

3. vaadin.elements.grid.GridTableScrollBehavior in vaadin-grid-table-edge-behavior.html is not defined, should it be vaadin.elements.grid.TableScrollBehavior?

Most of the compile errors are gone. I'm still getting these:
```
error:   In bower_components/vaadin-grid/vaadin-grid-column.html: [unknown-polymer-behavior] - Unable to resolve behavior `vaadinGridColumnBaseBehavior`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/vaadin-grid-column.html: [unknown-polymer-behavior] - Unable to resolve behavior `vaadinGridColumnBehaviorImpl`. Did you import it? Is it annotated with @polymerBehavior?
error:   In bower_components/vaadin-grid/vaadin-grid-column.html: [unknown-polymer-behavior] - Unable to resolve behavior `vaadinGridColumnBaseBehavior`. Did you import it? Is it annotated with @polymerBehavior?
```

--- a/iron-list-behavior.html
+++ b/iron-list-behavior.html
@@ -1,3 +1,7 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-resizable-behavior/iron-resizable-behavior.html">
+<link rel="import" href="../iron-scroll-target-behavior/iron-scroll-target-behavior.html">
+
 <script>
   window.vaadin = window.vaadin || {};
   vaadin.elements = vaadin.elements || {};
@@ -6,7 +10,7 @@
   /**
    * @polymerBehavior vaadin.elements.grid.IronListBehavior
    */
-  vaadin.elements.grid.IronListBehavior = [(function() {
+  vaadin.elements.grid.IronListBehaviorImpl = (function() {
 
     var IOS = navigator.userAgent.match(/iP(?:hone|ad;(?: U;)? CPU) OS (\d+)/);
     var IOS_TOUCH_SCROLLING = IOS && IOS[1] >= 8;
@@ -738,10 +742,15 @@
         return idx >= this.firstVisibleIndex && idx <= this.lastVisibleIndex;
       }
     });
+  })();
 
-  })(),
+  /**
+   * @polymerBehavior vaadin.elements.grid.IronListBehavior
+   */
+  vaadin.elements.grid.IronListBehavior = [
     Polymer.Templatizer,
     Polymer.IronResizableBehavior,
-    Polymer.IronScrollTargetBehavior
+    Polymer.IronScrollTargetBehavior,
+    vaadin.elements.grid.IronListBehaviorImpl
   ];
 </script>

--- a/iron-list-behavior.html
+++ b/iron-list-behavior.html
@@ -8,7 +8,7 @@
   vaadin.elements.grid = vaadin.elements.grid || {};
 
   /**
-   * @polymerBehavior vaadin.elements.grid.IronListBehavior
+   * @polymerBehavior vaadin.elements.grid.IronListBehaviorImpl
    */
   vaadin.elements.grid.IronListBehaviorImpl = (function() {
 

--- a/vaadin-grid-column.html
+++ b/vaadin-grid-column.html
@@ -13,9 +13,9 @@
   vaadin.elements.grid = vaadin.elements.grid || {};
 
   /**
-   * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBaseBehavior
+   * @polymerBehavior vaadin.elements.grid.ColumnBaseBehavior
    */
-  vaadin.elements.grid.vaadinGridColumnBaseBehavior = {
+  vaadin.elements.grid.ColumnBaseBehavior = {
     properties: {
       headerTemplate: {
         type: Object,
@@ -92,9 +92,9 @@
   };
 
   /**
-   * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBehaviorImpl
+   * @polymerBehavior vaadin.elements.grid.ColumnBehaviorImpl
    */
-  vaadin.elements.grid.vaadinGridColumnBehaviorImpl = {
+  vaadin.elements.grid.ColumnBehaviorImpl = {
     properties: {
       width: {
         type: String,
@@ -135,23 +135,23 @@
   }
 
   /**
-   * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBehavior
+   * @polymerBehavior vaadin.elements.grid.ColumnBehavior
    */
-  vaadin.elements.grid.vaadinGridColumnBehavior = [
-    vaadin.elements.grid.vaadinGridColumnBaseBehavior,
-    vaadin.elements.grid.vaadinGridColumnBehaviorImpl
+  vaadin.elements.grid.ColumnBehavior = [
+    vaadin.elements.grid.ColumnBaseBehavior,
+    vaadin.elements.grid.ColumnBehaviorImpl
   ];
 
   Polymer({
     is: 'vaadin-grid-column',
 
-    behaviors: [vaadin.elements.grid.vaadinGridColumnBehavior]
+    behaviors: [vaadin.elements.grid.ColumnBehavior]
   });
 
   Polymer({
     is: 'vaadin-grid-column-group',
 
-    behaviors: [vaadin.elements.grid.vaadinGridColumnBaseBehavior],
+    behaviors: [vaadin.elements.grid.ColumnBaseBehavior],
 
     properties: {
       _childColumns: {

--- a/vaadin-grid-column.html
+++ b/vaadin-grid-column.html
@@ -7,271 +7,275 @@
 </dom-module>
 
 <script>
-  (function() {
 
-    var vaadinGridColumnBaseBehavior = {
-      properties: {
-        headerTemplate: {
-          type: Object,
-          value: function() {
-            return this._findTemplate('template.header') || null;
-          }
-        },
-        footerTemplate: {
-          type: Object,
-          value: function() {
-            return this._findTemplate('template.footer') || null;
-          }
-        },
-        instances: {
-          type: Array,
-          value: function() {
-            return [];
-          }
-        },
+  window.vaadin = window.vaadin || {};
+  vaadin.elements = vaadin.elements || {};
+  vaadin.elements.grid = vaadin.elements.grid || {};
 
-        frozen: {
-          type: Boolean,
-          notify: true
-        },
-
-        _lastFrozen: {
-          type: Boolean,
-          notify: true
+  /**
+   * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBaseBehavior
+   */
+  vaadin.elements.grid.vaadinGridColumnBaseBehavior = {
+    properties: {
+      headerTemplate: {
+        type: Object,
+        value: function() {
+          return this._findTemplate('template.header') || null;
+        }
+      },
+      footerTemplate: {
+        type: Object,
+        value: function() {
+          return this._findTemplate('template.footer') || null;
+        }
+      },
+      instances: {
+        type: Array,
+        value: function() {
+          return [];
         }
       },
 
-      observers: [
-        '_footerTemplateChanged(footerTemplate)',
-        '_headerTemplateChanged(headerTemplate)',
-        '_lastFrozenChanged(_lastFrozen)'
-      ],
-
-      _selectFirstTemplate: function(selector) {
-        return Polymer.dom(this).querySelectorAll(selector).filter(function(el) {
-          return el.parentElement === this;
-        }.bind(this))[0]
+      frozen: {
+        type: Boolean,
+        notify: true
       },
 
-      _findTemplate: function(selector) {
-        var template = this._selectFirstTemplate(selector);
-        if (template) {
-          if (this.dataHost) {
-            // set dataHost to the context where template has been defined
-            template._rootDataHost = this.dataHost._rootDataHost || this.dataHost;
-          }
-        }
-        return template;
-      },
-
-      _headerTemplateChanged: function(headerTemplate) {
-        this.fire('property-changed', {path: 'headerTemplate', value: headerTemplate});
-      },
-
-      _footerTemplateChanged: function(footerTemplate) {
-        this.fire('property-changed', {path: 'footerTemplate', value: footerTemplate});
-      },
-
-      _flexChanged: function(flex) {
-        this.fire('property-changed', {path: 'flex', value: flex});
-      },
-
-      _widthChanged: function(width) {
-        this.fire('property-changed', {path: 'width', value: width});
-      },
-
-      _lastFrozenChanged: function(lastFrozen) {
-        this.fire('property-changed', {path: 'lastFrozen', value: lastFrozen});
+      _lastFrozen: {
+        type: Boolean,
+        notify: true
       }
-    };
+    },
 
-    var vaadinGridColumnBehaviorImpl = {
-      properties: {
-        width: {
-          type: String,
-          value: '100px'
-        },
+    observers: [
+      '_footerTemplateChanged(footerTemplate)',
+      '_headerTemplateChanged(headerTemplate)',
+      '_lastFrozenChanged(_lastFrozen)'
+    ],
 
-        flex: {
-          type: Number,
-          value: 1
-        },
+    _selectFirstTemplate: function(selector) {
+      return Polymer.dom(this).querySelectorAll(selector).filter(function(el) {
+        return el.parentElement === this;
+      }.bind(this))[0]
+    },
 
-        template: {
-          type: Object,
-          value: function() {
-            return this._findTemplate('template:not(.header):not(.footer)');
-          }
-        },
-
-        hidden: {
-          type: Boolean
+    _findTemplate: function(selector) {
+      var template = this._selectFirstTemplate(selector);
+      if (template) {
+        if (this.dataHost) {
+          // set dataHost to the context where template has been defined
+          template._rootDataHost = this.dataHost._rootDataHost || this.dataHost;
         }
-      },
-
-      observers: ['_flexChanged(flex)', '_widthChanged(width)', '_templateChanged(template)', '_frozenChanged(frozen)', '_hiddenChanged(hidden)'],
-
-      _frozenChanged: function(frozen) {
-        this.fire('property-changed', {path: 'frozen', value: frozen});
-      },
-
-      _templateChanged: function(template) {
-        // We bubble false for optimisation
-        this.fire('property-changed', {path: 'template', value: template}, {bubbles: false});
-      },
-
-      _hiddenChanged: function(hidden) {
-        this.fire('property-changed', {path: 'hidden', value: hidden});
       }
+      return template;
+    },
+
+    _headerTemplateChanged: function(headerTemplate) {
+      this.fire('property-changed', {path: 'headerTemplate', value: headerTemplate});
+    },
+
+    _footerTemplateChanged: function(footerTemplate) {
+      this.fire('property-changed', {path: 'footerTemplate', value: footerTemplate});
+    },
+
+    _flexChanged: function(flex) {
+      this.fire('property-changed', {path: 'flex', value: flex});
+    },
+
+    _widthChanged: function(width) {
+      this.fire('property-changed', {path: 'width', value: width});
+    },
+
+    _lastFrozenChanged: function(lastFrozen) {
+      this.fire('property-changed', {path: 'lastFrozen', value: lastFrozen});
     }
+  };
 
-    window.vaadin = window.vaadin || {};
-    vaadin.elements = vaadin.elements || {};
-    vaadin.elements.grid = vaadin.elements.grid || {};
-    /**
-     * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBehavior
-     */
-    vaadin.elements.grid.vaadinGridColumnBehavior = [
-      vaadinGridColumnBaseBehavior,
-      vaadinGridColumnBehaviorImpl
-    ];
+  /**
+   * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBehaviorImpl
+   */
+  vaadin.elements.grid.vaadinGridColumnBehaviorImpl = {
+    properties: {
+      width: {
+        type: String,
+        value: '100px'
+      },
 
-    Polymer({
-      is: 'vaadin-grid-column',
+      flex: {
+        type: Number,
+        value: 1
+      },
 
-      behaviors: [vaadin.elements.grid.vaadinGridColumnBehavior]
-    });
-
-    Polymer({
-      is: 'vaadin-grid-column-group',
-
-      behaviors: [vaadinGridColumnBaseBehavior],
-
-      properties: {
-        _childColumns: {
-          value: function() {
-            return Polymer.dom(this).querySelectorAll('vaadin-grid-column, vaadin-grid-selection-column');
-          }
-        },
-
-        flex: {
-          type: Number,
-          readOnly: true
-        },
-
-        width: {
-          type: String,
-          readOnly: true
-        },
-
-        _visibleChildColumns: Array,
-
-        hidden: {
-          type: Boolean,
-          notify: true,
-          readOnly: true
-        },
-
-        colSpan: {
-          type: Number,
-          notify: true,
-          readOnly: true
+      template: {
+        type: Object,
+        value: function() {
+          return this._findTemplate('template:not(.header):not(.footer)');
         }
       },
 
-      observers: ['_updateVisibleChildColumns(_childColumns)',
-      '_childColumnsChanged(_childColumns)',
-      '_flexChanged(flex)',
-      '_widthChanged(width)',
-      '_frozenChanged(_childColumns, frozen)',
-      '_hiddenChanged(hidden)',
-      '_visibleChildColumnsChanged(_visibleChildColumns)',
-      '_colSpanChanged(colSpan)'],
-
-      listeners: {
-        'property-changed': '_columnPropChanged'
-      },
-
-      attached: function() {
-        this._updateFlexAndWidth(this._visibleChildColumns);
-        this._addNodeObserver();
-      },
-
-      detached: function() {
-        Polymer.dom(this).unobserveNodes(this._observer);
-      },
-
-      _columnPropChanged: function(e) {
-        if (e.detail.path === 'hidden') {
-          this._updateVisibleChildColumns(this._childColumns);
-        }
-
-        if (/flex|width|hidden|_childColumns/.test(e.detail.path)) {
-          this._updateFlexAndWidth(this._visibleChildColumns);
-        }
-
-        if (e.detail.path === 'frozen') {
-          this.frozen = e.detail.value;
-        }
-
-        if (e.detail.path === 'lastFrozen') {
-          this._lastFrozen = e.detail.value;
-        }
-      },
-
-      _updateVisibleChildColumns: function(childColumns) {
-        this._visibleChildColumns = childColumns.filter(function(col) {
-          return !col.hidden;
-        });
-      },
-
-      _childColumnsChanged: function(childColumns) {
-        this.fire('property-changed', {path: '_childColumns', value: childColumns});
-      },
-
-      _updateFlexAndWidth: function(visibleChildColumns) {
-        this._setWidth('calc(' + visibleChildColumns.reduce(function(prev, curr) {
-          return prev += ' + ' + (curr.width || '0').replace('calc', '');
-        }, '').substring(3) + ')');
-
-        this._setFlex(visibleChildColumns.reduce(function(prev, curr) {
-          return prev + curr.flex;
-        }, 0));
-      },
-
-      _frozenChanged: function(childColumns, frozen) {
-        childColumns.forEach(function(col) {
-          col.frozen = frozen;
-        });
-        this.fire('property-changed', {path: 'frozen', value: frozen});
-      },
-
-      _hiddenChanged: function(hidden) {
-        this.fire('property-changed', {path: 'hidden', value: hidden});
-      },
-
-      _visibleChildColumnsChanged: function(visibleChildColumns) {
-        this._setColSpan(visibleChildColumns.length);
-        this._setHidden(this.colSpan === 0);
-      },
-
-      _colSpanChanged: function(colSpan) {
-        this.fire('property-changed', {path: 'colSpan', value: colSpan});
-      },
-
-      _addNodeObserver: function() {
-        this._observer = Polymer.dom(this).observeNodes(function(info) {
-          var columns = function(node) {
-            return (node.nodeType === Node.ELEMENT_NODE && node.localName.indexOf('vaadin-grid-column') === 0);
-          };
-          if (info.addedNodes.filter(columns).length > 0 ||
-          info.removedNodes.filter(columns).length > 0) {
-            this._childColumns = Polymer.dom(this).querySelectorAll('vaadin-grid-column');
-          }
-        }.bind(this));
+      hidden: {
+        type: Boolean
       }
-    });
+    },
 
-  }());
+    observers: ['_flexChanged(flex)', '_widthChanged(width)', '_templateChanged(template)', '_frozenChanged(frozen)', '_hiddenChanged(hidden)'],
+
+    _frozenChanged: function(frozen) {
+      this.fire('property-changed', {path: 'frozen', value: frozen});
+    },
+
+    _templateChanged: function(template) {
+      // We bubble false for optimisation
+      this.fire('property-changed', {path: 'template', value: template}, {bubbles: false});
+    },
+
+    _hiddenChanged: function(hidden) {
+      this.fire('property-changed', {path: 'hidden', value: hidden});
+    }
+  }
+
+  /**
+   * @polymerBehavior vaadin.elements.grid.vaadinGridColumnBehavior
+   */
+  vaadin.elements.grid.vaadinGridColumnBehavior = [
+    vaadin.elements.grid.vaadinGridColumnBaseBehavior,
+    vaadin.elements.grid.vaadinGridColumnBehaviorImpl
+  ];
+
+  Polymer({
+    is: 'vaadin-grid-column',
+
+    behaviors: [vaadin.elements.grid.vaadinGridColumnBehavior]
+  });
+
+  Polymer({
+    is: 'vaadin-grid-column-group',
+
+    behaviors: [vaadin.elements.grid.vaadinGridColumnBaseBehavior],
+
+    properties: {
+      _childColumns: {
+        value: function() {
+          return Polymer.dom(this).querySelectorAll('vaadin-grid-column, vaadin-grid-selection-column');
+        }
+      },
+
+      flex: {
+        type: Number,
+        readOnly: true
+      },
+
+      width: {
+        type: String,
+        readOnly: true
+      },
+
+      _visibleChildColumns: Array,
+
+      hidden: {
+        type: Boolean,
+        notify: true,
+        readOnly: true
+      },
+
+      colSpan: {
+        type: Number,
+        notify: true,
+        readOnly: true
+      }
+    },
+
+    observers: ['_updateVisibleChildColumns(_childColumns)',
+    '_childColumnsChanged(_childColumns)',
+    '_flexChanged(flex)',
+    '_widthChanged(width)',
+    '_frozenChanged(_childColumns, frozen)',
+    '_hiddenChanged(hidden)',
+    '_visibleChildColumnsChanged(_visibleChildColumns)',
+    '_colSpanChanged(colSpan)'],
+
+    listeners: {
+      'property-changed': '_columnPropChanged'
+    },
+
+    attached: function() {
+      this._updateFlexAndWidth(this._visibleChildColumns);
+      this._addNodeObserver();
+    },
+
+    detached: function() {
+      Polymer.dom(this).unobserveNodes(this._observer);
+    },
+
+    _columnPropChanged: function(e) {
+      if (e.detail.path === 'hidden') {
+        this._updateVisibleChildColumns(this._childColumns);
+      }
+
+      if (/flex|width|hidden|_childColumns/.test(e.detail.path)) {
+        this._updateFlexAndWidth(this._visibleChildColumns);
+      }
+
+      if (e.detail.path === 'frozen') {
+        this.frozen = e.detail.value;
+      }
+
+      if (e.detail.path === 'lastFrozen') {
+        this._lastFrozen = e.detail.value;
+      }
+    },
+
+    _updateVisibleChildColumns: function(childColumns) {
+      this._visibleChildColumns = childColumns.filter(function(col) {
+        return !col.hidden;
+      });
+    },
+
+    _childColumnsChanged: function(childColumns) {
+      this.fire('property-changed', {path: '_childColumns', value: childColumns});
+    },
+
+    _updateFlexAndWidth: function(visibleChildColumns) {
+      this._setWidth('calc(' + visibleChildColumns.reduce(function(prev, curr) {
+        return prev += ' + ' + (curr.width || '0').replace('calc', '');
+      }, '').substring(3) + ')');
+
+      this._setFlex(visibleChildColumns.reduce(function(prev, curr) {
+        return prev + curr.flex;
+      }, 0));
+    },
+
+    _frozenChanged: function(childColumns, frozen) {
+      childColumns.forEach(function(col) {
+        col.frozen = frozen;
+      });
+      this.fire('property-changed', {path: 'frozen', value: frozen});
+    },
+
+    _hiddenChanged: function(hidden) {
+      this.fire('property-changed', {path: 'hidden', value: hidden});
+    },
+
+    _visibleChildColumnsChanged: function(visibleChildColumns) {
+      this._setColSpan(visibleChildColumns.length);
+      this._setHidden(this.colSpan === 0);
+    },
+
+    _colSpanChanged: function(colSpan) {
+      this.fire('property-changed', {path: 'colSpan', value: colSpan});
+    },
+
+    _addNodeObserver: function() {
+      this._observer = Polymer.dom(this).observeNodes(function(info) {
+        var columns = function(node) {
+          return (node.nodeType === Node.ELEMENT_NODE && node.localName.indexOf('vaadin-grid-column') === 0);
+        };
+        if (info.addedNodes.filter(columns).length > 0 ||
+        info.removedNodes.filter(columns).length > 0) {
+          this._childColumns = Polymer.dom(this).querySelectorAll('vaadin-grid-column');
+        }
+      }.bind(this));
+    }
+  });
 
 </script>

--- a/vaadin-grid-selection-column.html
+++ b/vaadin-grid-selection-column.html
@@ -16,7 +16,7 @@
   Polymer({
     is: 'vaadin-grid-selection-column',
 
-    behaviors: [vaadin.elements.grid.vaadinGridColumnBehavior],
+    behaviors: [vaadin.elements.grid.ColumnBehavior],
 
     properties: {
       width: {

--- a/vaadin-grid-table-cell.html
+++ b/vaadin-grid-table-cell.html
@@ -1,3 +1,7 @@
+
+<link rel="import" href="../polymer/polymer.html">
+
+
 <link rel="import" href="vaadin-grid-cell-click-behavior.html">
 
 <dom-module id="vaadin-grid-table-cell"></dom-module>

